### ISSUE TITLE
Raspberry Pi 3

### ISSUE THREAD

**Issue by micrcx** [otcshare/iotivity-node#97]
Is it possible to install nodejs iotivity on Raspberry PI 3? What do need to do for this? You can advise something? Maybe there are ready-made solution?

**Comment by iamsanjeev** [comment]
I'd imagine it should work out of the box. I havent tried it though.
scons build for resource layer should be invoked with TARGET_ARCH=armv7a maybe.
Other than that the process is straight forward. Assuming Raspbian the following steps should work.

1. npm install iotivity-node
2. node node_modules/iotivity-node/js/high-level-server-example.js
3. node node_modules/iotivity-node/js/high-level-client-example.js

**Comment by rzr** [comment]
I think it should be possible , later I'll try to cook a Tizen PI3 image if you can test it ?

**Comment by rzr** [comment]
Those are work in progress, but you rebuild an image with iotivity 1.2.0 and iotivity-node  ,
just change MACHINE=raspberrypi3

https://wiki.iotivity.org/automotive#work_in_progress_120

I am building for pi2 at the moment

**Comment by mjkoster** [comment]
I am trying to use Ubuntu on Raspberry Pi 3 and getting the -m32 flag set even though verbose output shows that it seems to correctly identifies arm architecture:

Checking for C function strptime()... no
scons: done reading SConscript files.
scons: Building targets ...
Compiling out/linux/x86/release/resource/csdk/logger/src/logger.o
gcc: error: unrecognized command line option '-m32'
scons: *** [out/linux/x86/release/resource/csdk/logger/src/logger.o] Error 1
scons: building terminated because of errors.

I'm installing by using:
npm install iot-rest-api-server 
and having it install iotivity-node and build iotivity as dependencies

I'm not sure what to do; it seems like it should "just work"...

**Comment by rzr** [comment]
out/linux/x86 is not expected for pi3 it should be out/linux/arm

**Comment by gabrielschulhof** [comment]
@mjkoster @micrcx is iotivity-node installation working for you now on Raspberry Pi 3?

**Comment by mjkoster** [comment]
It was working in July for the W3C Plugtest, but I haven't tried after the last update to the iotivity 1.3 version. I was just about to test it all again in preparation for more development work, so will get back to you with the results.

**Comment by rzr** [comment]
@kelsayed , Can you build regular iotivity on it ?

FYI, I was able to build snapshots of master branch for raspbian and can share binaries at:
https://dl.bintray.com/rzr/raspbian-9-armhf/

**Comment by kelsayed** [comment]
Yes, regular iotivity compiles but compiling iotivity from within iotivity-node need to be hacked for some .h files were not visible. 

Which of these .deb correspond to iotivity-node binaries?

**Comment by rzr** [comment]
none this just iotivity, I havent looked at i-node debian packaging to not overlap with NPM one.
